diff --git a/lmkd.cpp b/lmkd.cpp
index 926a1d0..f3f4085 100644
--- a/lmkd.cpp
+++ b/lmkd.cpp
@@ -2570,6 +2570,11 @@ static void mp_event_psi(int data, uint32_t events, struct polling_params *poll_
          * changing
          */
         goto no_kill;
+    } else if (workingset_refault_file < prev_workingset_refault) {
+        /* workingset_refault counter wrapped or is unavailable, reset baseline */
+        prev_workingset_refault = workingset_refault_file;
+        init_ws_refault = workingset_refault_file;
+        goto no_kill;
     }
 
     prev_workingset_refault = workingset_refault_file;
