diff --git a/libs/ui/GraphicBuffer.cpp b/libs/ui/GraphicBuffer.cpp
index 3732fee7f2..adb4ed1735 100644
--- a/libs/ui/GraphicBuffer.cpp
+++ b/libs/ui/GraphicBuffer.cpp
@@ -527,3 +527,5 @@ void GraphicBuffer::addDeathCallback(GraphicBufferDeathCallback deathCallback, v
 // ---------------------------------------------------------------------------
 
 }; // namespace android
+
+__strong_alias(_ZN7android13GraphicBufferC1EjjijjP13native_handleb,_ZN7android13GraphicBufferC1EjjijjjP13native_handleb);
